hal/inc/hal_platform_compat.h

```diff
@@ -22,10 +22,13 @@
 
 #if PLATFORM_ID <= PLATFORM_P1 || PLATFORM_ID == PLATFORM_NEWHAL
     #define HAL_PLATFORM_WIFI 1
+    // Just define it
+    #define HAL_PLATFORM_WIFI_SERIAL
```

Comment:
Seems confusing. Photon/P1 use SDIO.

Reply:
How about defining `HAL_PLATFORM_WIFI_SDIO` and asserting if `!defined HAL_PLATFORM_WIFI_SERIAL && !defined HAL_PLATFORM_WIFI_SDIO` in `hal_platform.h`

Reply:
There is already `HAL_PLATFORM_NCP_AT` as well.